### PR TITLE
Improve Framework detection and tests

### LINKED ISSUE #520: Angular no longer works with 2.2.0.CR1 (pnpm)
### Describe the bug

My start script is
```json
"start": "ng serve"
```
and I'm using Angular `16.2.6`.

I get the following error if I start the Quarkus dev server:
```java
2023-09-25T22:31:45.272Z INFO  [FrameworkType] (build-10) Quinoa detected 'ANGULAR' framework from package.json file.
2023-09-25T22:31:46.108Z INFO  [PackageManagerRunner] (build-11) Running Quinoa package manager live coding as a dev service: pnpm run start -- --disable-host-check
> fss-web@0.1.0 start C:\Users\ThoSap\IdeaProjects\fss-web\src\main\webui
> ng serve "--" "--disable-host-check"
Option '--' has been specified multiple times. The value '--disable-host-check' will be used.
Error: Schema validation failed with the following errors:
  Data path "" must NOT have additional properties().
?ELIFECYCLE? Command failed with exit code 1.
2023-09-25 22:32:48,506 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2023-09-25 22:32:48,801 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkiverse.quinoa.deployment.ForwardedDevProcessor#prepareDevService threw an exception: java.lang.RuntimeException: java.lang.RuntimeException: Quinoa package manager live coding port 4200 is still not listening after the checkTimeout.
	at io.quarkiverse.quinoa.deployment.ForwardedDevProcessor.prepareDevService(ForwardedDevProcessor.java:162)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: Quinoa package manager live coding port 4200 is still not listening after the checkTimeout.
	at io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner.dev(PackageManagerRunner.java:161)
	at io.quarkiverse.quinoa.deployment.ForwardedDevProcessor.prepareDevService(ForwardedDevProcessor.java:136)
	... 11 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:336)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:253)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:113)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:438)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:56)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkiverse.quinoa.deployment.ForwardedDevProcessor#prepareDevService threw an exception: java.lang.RuntimeException: java.lang.RuntimeException: Quinoa package manager live coding port 4200 is still not listening after the checkTimeout.
	at io.quarkiverse.quinoa.deployment.ForwardedDevProcessor.prepareDevService(ForwardedDevProcessor.java:162)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: Quinoa package manager live coding port 4200 is still not listening after the checkTimeout.
	at io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner.dev(PackageManagerRunner.java:161)
	at io.quarkiverse.quinoa.deployment.ForwardedDevProcessor.prepareDevService(ForwardedDevProcessor.java:136)
	... 11 more

	at io.quarkus.builder.Execution.run(Execution.java:123)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:332)
	... 9 more
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: Quinoa package manager live coding port 4200 is still not listening after the checkTimeout.
	at io.quarkiverse.quinoa.deployment.ForwardedDevProcessor.prepareDevService(ForwardedDevProcessor.java:162)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: Quinoa package manager live coding port 4200 is still not listening after the checkTimeout.
	at io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner.dev(PackageManagerRunner.java:161)
	at io.quarkiverse.quinoa.deployment.ForwardedDevProcessor.prepareDevService(ForwardedDevProcessor.java:136)
	... 11 more
```

### Quinoa version

2.2.0.CR1

### Quarkus version

3.4.1

### Build / Runtime

Angular

### Package Manager

PNPM

### Steps to reproduce the behavior

Modify the package.json start script as follows:
```json
"start": "ng serve"
```

### Expected behavior

Start the Quarkus dev server, and the Angular dev server

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #520 #519 

- Clean frameworks doc page (some frameworks were too weird to be considered as detected)
- I added tests to make sure the framework detection is working
- if we consider the package.json is customized we don't auto add the args (--hmr for angular, ...)
- for pnpm we remove the '--' to make it work

### CODE REVIEW

**Comment by ia3andy** [comment]
@melloware this is another large PR but I think it is another good bunch of improvements for detection and logging

**Comment by ia3andy** [comment]
This is what the test run prints:
```
2023-09-27 18:12:43,008 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing react-create-app-exact
2023-09-27 18:12:43,066 DEBUG [io.qua.qui.dep.fra.FrameworkType] (main) Detected framework with dev command perfect match: REACT
2023-09-27 18:12:43,135 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

2023-09-27 18:12:43,150 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing react-create-app-different-dev-script
2023-09-27 18:12:43,152 DEBUG [io.qua.qui.dep.fra.FrameworkType] (main) Detected framework with dev command perfect match: REACT
2023-09-27 18:12:43,152 WARN  [io.qua.qui.dep.fra.FrameworkType] (main) REACT framework typically defines a 'start` script in package.json file but found 'dev' instead.
2023-09-27 18:12:43,152 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

2023-09-27 18:12:43,156 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing react-create-app-different-dev-script-envs
2023-09-27 18:12:43,159 WARN  [io.qua.qui.dep.fra.FrameworkType] (main) REACT framework typically defines a 'start` script in package.json file but found 'dev' instead.
2023-09-27 18:12:43,159 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

2023-09-27 18:12:43,165 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing react-create-app-multiple-ok
2023-09-27 18:12:43,167 DEBUG [io.qua.qui.dep.fra.FrameworkType] (main) Detected framework with dev command perfect match: REACT
2023-09-27 18:12:43,167 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

2023-09-27 18:12:43,171 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing react-create-app-multiple-ok-guess
2023-09-27 18:12:43,173 ERROR [io.qua.qui.dep.fra.FrameworkType] (main) Framework detection failed: It is probably REACT framework which typically defines a 'start' script in package.json but found multiple other dev scripts instead (dev, start).
2023-09-27 18:12:43,173 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

2023-09-27 18:12:43,178 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing react-create-app-multiple-ok-guess-command
2023-09-27 18:12:43,180 DEBUG [io.qua.qui.dep.fra.FrameworkType] (main) Detected framework with dev command perfect match: REACT
2023-09-27 18:12:43,181 WARN  [io.qua.qui.dep.fra.FrameworkType] (main) REACT framework typically defines a 'start` script in package.json file but found 'develop' instead.
2023-09-27 18:12:43,181 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

2023-09-27 18:12:43,185 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing react-create-app-multiple-nok
2023-09-27 18:12:43,187 ERROR [io.qua.qui.dep.fra.FrameworkType] (main) Framework detection failed: It is probably REACT framework which typically defines a 'start' script in package.json but found multiple other dev scripts instead (dev, develop).
2023-09-27 18:12:43,187 ERROR [io.qua.qui.dep.fra.FrameworkType] (main) Framework detection failed: It is probably REACT framework which typically defines a 'start' script in package.json but found multiple other dev scripts instead (dev, develop).
2023-09-27 18:12:43,187 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

2023-09-27 18:12:43,191 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing angular-exact
2023-09-27 18:12:43,194 DEBUG [io.qua.qui.dep.fra.FrameworkType] (main) Detected framework with dev command perfect match: ANGULAR
2023-09-27 18:12:43,194 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

2023-09-27 18:12:43,198 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing angular-envs
2023-09-27 18:12:43,200 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

2023-09-27 18:12:43,203 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) Testing angular-args
2023-09-27 18:12:43,204 INFO  [io.qua.qui.dep.fra.FrameworkTypeTest] (main) 

```

File: deployment/src/test/java/io/quarkiverse/quinoa/deployment/framework/FrameworkTypeTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.assertj.core.api.Assertions;
+import org.jboss.logging.Logger;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class FrameworkTypeTest {
+
+    private static final Logger LOG = Logger.getLogger(FrameworkTypeTest.class);
+
+    @ParameterizedTest
```
Comment (melloware) [inline deployment/src/test/java/io/quarkiverse/quinoa/deployment/framework/FrameworkTypeTest.java:18]: Nice use of parameterized test!

File: deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/FrameworkType.java [hunk 2]
```diff
@@ -6,11 +6,14 @@
...
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.AbstractMap.SimpleImmutableEntry;
 import java.util.Arrays;
 import java.util.Locale;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
+import java.util.StringJoiner;
```
Comment (melloware) [inline deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/FrameworkType.java:16]: I just learned something new. I have been using Commons Lang `StringUtils.join` for years didn't know it was in the JDK now!

**Review by melloware — APPROVED** [review]

**Pull request merged**